gui/gui/src/TGSpeedo.cxx

```diff
@@ -300,6 +315,17 @@ Bool_t TGSpeedo::HandleButton(Event_t *event)
    return kTRUE;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Change the circular buffer size (used for the automatic mean calculation).
+
+void TGSpeedo::SetBufferSize(Int_t size)
+{
+   fBufferSize = size;
+   if (fBufferCount > fBufferSize)
+      fBufferCount = fBufferSize;
+   fBuffer.resize(fBufferSize);
```

Comment:
Maybe it would be cleaner to do:
```
fBuffer.clear();
fBuffer.reserve(fBufferSize);
fBufferCount = 0;
```

Reply:
No, it has to be `resize()`, to properly resize it (`reserve` only allocates the memory and doesn't affect `size()`)

Reply:
Yes, but that's why I was suggesting to use push_back for the first N entries, so that the mean value in the beginning is not distorted.